types/wlr_surface.c

```diff
@@ -91,11 +91,33 @@ static void surface_set_input_region(struct wl_client *client,
 	}
 }
 
+static void wlr_surface_get_effective_size(struct wlr_surface *surface,
+		int swidth, int sheight, int *width, int *height) {
+	int scale = surface->current.scale;
+	enum wl_output_transform transform = surface->current.transform;
+	int _width = swidth / scale;
+	int _height = sheight / scale;
```

Comment:
I don't think the scale should be used here, no? This is also meant to be public, and swidth/sheight should be removed. If you need those to work with a partially committed surface, then keep this function around but add a new public function that just grabs it from current.